### PR TITLE
MP-9075 NOX - Refactor EntityControllerGenerator to use Scriban template - Migrating endpoints for Entity

### PR DESCRIPTION
**Pull request by emir-sehovic1**

### CODE REVIEW

File: src/Nox.Generator/Common/TemplateScriptsBridges/NoxSolutionBridge.cs [hunk 1]
```diff
@@ -43,6 +45,17 @@ public static void AddFunctions(TemplateContext context, NoxSolution noxSolution
         scriptObject8.Import("IsValueType", new Func<string,
             bool>(type => Type.GetType(type)?.IsValueType ?? false));
 
+        var scriptObject9 = new ScriptObject();
```
Comment (rochar) [inline src/Nox.Generator/Common/TemplateScriptsBridges/NoxSolutionBridge.cs:48]: see https://github.com/scriban/scriban/blob/master/doc/builtins.md#string-functions, thera are scriban native functions, we may remove this

File: src/Nox.Generator/Common/TemplateScriptsBridges/NoxSolutionBridge.cs [hunk 2]
```diff
@@ -43,6 +45,17 @@ public static void AddFunctions(TemplateContext context, NoxSolution noxSolution
...
+        scriptObject9.Import("ToLowerFirstChar", new Func<string, string>(
+            input => input.ToLowerFirstChar()));
+
+        var scriptObject10 = new ScriptObject();
+        scriptObject10.Import("GetPrimaryKeys", PrimaryKeysQuery);
+
+        var scriptObject11 = new ScriptObject();
```
Comment (rochar) [inline src/Nox.Generator/Common/TemplateScriptsBridges/NoxSolutionBridge.cs]: this is only related with API Generator and the `PrimaryKeysFromRoute `can be done directly on the template so its clear what we are generating

and the same for this function `PrimaryKeysQuery`

Comment (rochar) [reply src/Nox.Generator/Common/TemplateScriptsBridges/NoxSolutionBridge.cs]: @emir-sehovic1  To keep it simple and move forward with the PR, lets add the result of this 2 function to the generator class, so its only available for this template

```
  new TemplateCodeBuilder(context, codeGeneratorState)
      .WithClassName($"{entity.PluralName}Controller")
      .WithFileNamePrefix("Presentation.Api.OData")
      .WithFileNameSuffix("Entity")
      .WithObject("entity", entity)
     .WithObject("PrimaryKeysQuery", GetPrimaryKeysQuery())
    ...
```

File: src/Nox.Generator/Common/TemplateScriptsBridges/NoxSolutionBridge.cs (hunk 1 above)
Comment (rochar) [reply src/Nox.Generator/Common/TemplateScriptsBridges/NoxSolutionBridge.cs:48]: that way we can remove the link to string build extensions

**Review by rochar — CHANGES_REQUESTED** [review]
looks good @emir-sehovic1 , juts two comments to considerer

**Review by rochar — APPROVED** [review]

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Pull request merged**